controllers/space/space_controller.go

```diff
@@ -308,6 +363,22 @@ func (r *Reconciler) setStatusTerminatingFailed(logger logr.Logger, space *toolc
 	return cause
 }
 
+func (r *Reconciler) setStatusChangeFailed(logger logr.Logger, space *toolchainv1alpha1.Space, cause error) error {
```

Comment:
I'm a bit confused by the name `setStatusChangeFailed`. Since it is setting the ready condition to false with reason `SpaceStatusUpdateFailedReason` then would it be better to rename it to `setStatusUpdateFailed`

Reply:
this code doesn't exist anymore